test/gtest/common/mem_buffer.cc

```diff
@@ -238,14 +244,14 @@ void *mem_buffer::allocate(size_t size, ucs_memory_type_t mem_type, bool async)
         return ptr;
 #if HAVE_CUDA
     case UCS_MEMORY_TYPE_CUDA:
-#if CUDA_VERSION >= 11020
+#if CUDART_VERSION >= 11020
```

Comment:
maybe move this preprocessor check inside if (async) condition, to ensure no fallback will happen?
smth like
```c
if (async) {
#if CUDART_VERSION >= 11020
    CUDA_CALL(cudaMallocAsync(&ptr, size, 0), ": size=" << size);
    cudaStreamSynchronize(0);
#else
    UCS_TEST_ABORT(..);
#endif
}
```

Reply:
But this is already covered by the if block at line 232-237.

Reply:
right, that block could be removed to simplify the code

Reply:
Applied suggestion and removed extra code.